### ISSUE TITLE
[Question] Have default tasks for filetype

### ISSUE THREAD

**Issue by TornaxO7** [jedrzejboczar/toggletasks.nvim#7]
Hello!

## TL;DR
Is it possible to have a `toggletask.json` file for a specifique file type (like in `~/.config/nvim/ftplugin`) which has some default-tasks for all files with the same filetype?

## Example
I mainly use `cargo build`, `cargo run` and `cargo check` for *almost all* rust projects. I don't want to create for every project a `toggletask.json` which includes those three tasks. So I thought that it would be possbile to store a `toggletasks.json` in `~/.config/nvim/ftplugin/rust/toggletasks.json` which is gonna be loaded if I enter a rust file.

## What I have tried
I tried to add the following to `nvim/ftplugin/rust`:

`~/.config/nvim/lua/toggletasks_settings/settings.lua`:

<details>

```lua
local toggletasks = require("toggletasks")

toggletasks.setup {
    scan = {
        global_cwd = true,
        tab_cwd = false,
        win_cwd = false,
        lsp_root = false,
        dirs = {},
    },
    defaults = {
        close_on_exit = false,
        hidden = false,
    },
}

require("toggletasks_settings.telescope")
require("toggletasks_settings.keybindings")
require("toggletasks_settings.whichkey")
```

</details>

`~/.config/nvim/ftplugin/rust/toggletasks_settings.lua`:

<details>

```lua
local toggletasks = require("toggletasks")

toggletasks.setup {
    scan = {
        dirs = {"/home/tornax/.config/nvim/ftplugin/rust"}
    }
}
```

</details>

`~/.config/nvim/ftplugin/rust/toggletasks.json`:

<details>

```json
{
    "tasks": {
        {
            "name": "build",
            "cmd": "cargo build"
        },
        {
            "name": "check",
            "cmd": "cargo check"
        },
        {
            "name": "run",
            "cmd": "cargo run"
        }
    }
}
```

</details>

but toggletask didn't load my `~/.config/nvim/ftplugin/rust/toggletasks.json`.

**Comment by jedrzejboczar** [comment]
This was something I wanted to implement but I never actually did. 

Could you test the changes from https://github.com/jedrzejboczar/toggletasks.nvim/tree/common-tasks? This adds option to set `scan.rtp_ftplugin = true`, which should add all directories from runtimepath that match `ftplugin/<FILETYPE>` to the scanned directories. So you should be able to put a file in `$HOME/.config/nvim/ftplugin/rust/toggletasks.json` and it should be loaded.

I was also thinking of just accepting a `function(win)` as `scan.dirs`, which would allow for even more flexibility. And I also though about adding a `tasks` option to setup, where one could just define tasks as Lua table (and it could be a `function(win)`.

**Comment by TornaxO7** [comment]
> Could you test the changes from https://github.com/jedrzejboczar/toggletasks.nvim/tree/common-tasks? This adds option to set scan.rtp_ftplugin = true, which should add all directories from runtimepath that match ftplugin/<FILETYPE> to the scanned directories. So you should be able to put a file in $HOME/.config/nvim/ftplugin/rust/toggletasks.json and it should be loaded.

Good to know, I'll try that soon if I find some time!

**Comment by TornaxO7** [comment]
@jedrzejboczar Yay! It's working :)

**Comment by jedrzejboczar** [comment]
I merged all the changes and everything should now be described here https://github.com/jedrzejboczar/toggletasks.nvim#global-tasks.
